compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -469,24 +469,17 @@ impl<'a> Resolver<'a> {
             ResolutionError::ParamInNonTrivialAnonConst { name, is_type } => {
                 let mut err = self.session.struct_span_err(
                     span,
-                    "generic parameters must not be used inside of non-trivial constant values",
-                );
-                err.span_label(
-                    span,
-                    &format!(
-                        "non-trivial anonymous constants must not depend on the parameter `{}`",
-                        name
-                    ),
+                    "generic parameters must not be used inside const evaluations",
                 );
+                err.span_label(span, &format!("cannot perform const operation using `{}`", name));
 
                 if is_type {
-                    err.note("type parameters are currently not permitted in anonymous constants");
+                    err.note("type parameters may not be used in anonymous constants");
```

Comment:
I'm wondering whether we even need these special note for types: I think the main error message essentially already contains this information.

Reply:
I think it helps to explicitly mention that type params are not supported at all, so I would personally keep that note

Reply:
Maybe "const expressions" is clearer than "anonymous constants" in that case?

Reply:
yeah, I guess this is where most of the issue lies - at least for me. What do we call the different constants?

So it seems to me like we settled on `const operations` for everything which is not a param.
Do you want to use `const expression` for all constants - or at least for all constants used in types and repeat expressions? One issue for me is that type parameters can be used in associated constants which are not `anonymous` but might also be `expressions`.

Reply:
> One issue for me is that type parameters can be used in associated constants which are not anonymous but might also be expressions.

Yes, that is a little awkward. I'd be tempted to treat that as an exception to the rule, and just make a blanket statement here. Eventually, we'd hope to relax this, so using specific terminology like "anonymous" for an edge case like this might end up being more confusing.

Reply:
hmm, I guess that's also fine with me :+1: If we go and slowly push for const wf bounds in the future simply using "const expression" for all constants looks good to me, even if it is currently somewhat incorrect for associated constants